### PR TITLE
Ensuring background modules are started before foreground modules

### PR DESCRIPTION
**Pull request by stulees**

### CODE REVIEW

**Comment by stulees** [comment]
@tejohnso I've removed a unit test and had to disable an installer integration test due to no longer restarting a module once extracted. I've removed the module restart upon extraction because we need precedence to background modules always starting before foreground, as per the changes in this PR. Need a re-think on how we upgrade and restart modules with this rule now. Thoughts? @rodrigopavezi FYI

File: installer/main-controller.js [hunk 1]
```diff
@@ -370,7 +371,14 @@ module.exports = {
...
       }
     })
+    .then(()=>{
+      if (options && options.startBackground) {
+        moduleController.startBackgroundModules();
+      }
+    })
+    .then(()=>{})
     .then(()=>{log.all("install complete", "", 100);})
+    .then(()=>{})
```
Comment (tejohnso) [inline installer/main-controller.js]: What are these empty functions for?

File: installer/installer.js [hunk 2]
```diff
@@ -102,7 +101,8 @@ module.exports = {
         })
         .then(()=>{
           config.updateManifestVersion(module.name, module.version);
-          moduleController.restartModule(module.name);
+          //TODO: Need to revise where/when to restart an upgraded module based on background modules needing precedence to start first
+          // moduleController.restartModule(module.name);
```
Comment (tejohnso) [inline installer/installer.js]: If we don't start anything here, any new downloaded module versions will start up on the next restart. I think that's simplest and should be fine.

Comment (tejohnso) [reply installer/installer.js]: The whole background-foreground thing might be unnecessary too.  Modules know when they connect to LM. It should be reasonable for them (player) to hold off on sending whatever they need to send until after they've connected to LM.

**Pull request closed without merging**